client/src/core-components/autocomplete.js

```diff
@@ -242,7 +242,10 @@ class Autocomplete extends React.Component {
         let id = ++this.searchApiRequestId;
 
         getItemListFromQuery && getItemListFromQuery(query, blacklist.map(
-            item => {return item.isStaff !== undefined ? {isStaff: item.isStaff, id: item.id} : item.id}
+            item => {
+                this.setState({loading: true});
```

Comment:
this is called multiple times unnecessarily, also it doesn't make sense that it's on a map.
why not put it in the line before?